app/functoria_tool.ml

```diff
@@ -18,102 +18,186 @@ open Cmdliner
-        ~doc:(Fmt.strf "Colorize the output. $(docv) must be %s." alts)
-    in
-    Arg.(value & opt color None & doc)
-
-  let verbose =
-    let doc =
-      Arg.info ~docs:global_option_section ~doc:"Be verbose" ["verbose";"v"]
-    in
-    Arg.(value & flag_all doc)
-
-  let init_log = function
-    | []  -> Functoria_misc.Log.(set_level WARN)
-    | [_] -> Functoria_misc.Log.(set_level INFO)
-    | _   -> Functoria_misc.Log.(set_level DEBUG)
-
-  let init_format color =
-    let i = Terminfo.columns () in
-    Functoria_misc.Log.set_color color;
-    Format.pp_set_margin Format.std_formatter i;
-    Format.pp_set_margin Format.err_formatter i;
-    Fmt_tty.setup_std_outputs ?style_renderer:color ()
+let global_option_section = "COMMON OPTIONS"
+
+let help_sections = [
+  `S global_option_section;
+  `P "These options are common to all commands.";
+]
+
+(* Helpers *)
+let mk_flag ?(section=global_option_section) flags doc =
+  let doc = Arg.info ~docs:section ~doc flags in
+  Arg.(value & flag & doc)
+
+let term_info title ~doc ~man =
+  let man = man @ help_sections in
+  Term.info ~sdocs:global_option_section ~doc ~man title
+
+let no_opam =
+  mk_flag ["no-opam"]
+    "Do not manage the OPAM configuration. \
+     This will result in dependent libraries not being automatically \
+     installed during the configuration phase."
+
+let no_opam_version_check =
+  mk_flag ["no-opam-version-check"] "Bypass the OPAM version check."
+
+let no_depext =
+  mk_flag ["no-depext"] "Skip installation of external dependencies."
+
+let full_eval =
+  mk_flag ["eval"]
+    "Fully evaluate the graph before showing it. \
+     By default, only the keys that are given on the command line are \
+     evaluated."
+
+let dot =
+  mk_flag ["dot"]
+    "Output a dot description. \
+     If no output file is given,  it will display the dot file using the command \
+     given to $(b,--dot-command)."
+
+let dotcmd =
+  let doc =
+    Arg.info ~docs:global_option_section ~docv:"COMMAND" [ "dot-command" ]
+      ~doc:"Command used to show a dot file. This command should accept a \
+            dot file on its standard input."
+  in
+  Arg.(value & opt string "xdot" & doc)
+
+let file =
+  let doc =
+    Arg.info ~docs:global_option_section ~docv:"CONFIG_FILE" ["f"; "file"]
+      ~doc:"Configuration file. If not specified, the current directory will \
+            be scanned. If one file named $(b,config.ml) is found, that file \
+            will be used. If no files or multiple configuration files are \
+            found, this will result in an error unless one is explicitly \
+            specified on the command line."
+  in
+  Arg.(value & opt (some file) None & doc)
+
+let output =
+  let doc =
+    Arg.info ~docs:global_option_section ~docv:"FILE" ["o"; "output"]
+      ~doc:"File where to output description or dot representation."
+  in
+  Arg.(value & opt (some string) None & doc)
+
+let color =
```

Comment:
If you want to remove a few lines you can use [`Fmt_cli.style_renderer`](http://erratique.ch/software/fmt/doc/Fmt_cli.html#VALstyle_renderer) available from `fmt` 0.7.1 (you will need to add a constraint in the `opam` file)